src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -222,6 +225,106 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         Automation::AutomationProperties::SetFullDescription(UseDesktopImageCheckBox(), unbox_value<hstring>(backgroundImgCheckboxTooltip));
 
         Automation::AutomationProperties::SetName(DeleteButton(), RS_(L"Profile_DeleteButton/Text"));
+
+        if (!_FontList)
+        {
+            _UpdateFontList();
+        }
+    }
+
+    void Profiles::_UpdateFontList()
+    try
+    {
+        // initialize font list
+        _FontList = single_threaded_observable_vector<hstring>();
+
+        // get a DWriteFactory
+        ::IUnknown* unknownFactory;
+        THROW_IF_FAILED(DWriteCreateFactory(
+            DWRITE_FACTORY_TYPE_SHARED,
+            __uuidof(::IDWriteFactory),
+            &unknownFactory));
+
+        ::IDWriteFactory* dwriteFactory;
+        THROW_IF_FAILED(unknownFactory->QueryInterface(&dwriteFactory));
+
+        com_ptr<::IDWriteFactory> factory{};
+        factory.attach(dwriteFactory);
+
+        // get the font collection; subscribe to updates
+        com_ptr<::IDWriteFontCollection> fontCollection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(fontCollection.put(), TRUE));
+
+        for (UINT32 i = 0; i < fontCollection->GetFontFamilyCount(); ++i)
+        {
+            try
+            {
+                // get the font
+                com_ptr<::IDWriteFontFamily> fontFamily;
+                THROW_IF_FAILED(fontCollection->GetFontFamily(i, fontFamily.put()));
+
+                // get the font's localized names
+                com_ptr<::IDWriteLocalizedStrings> localizedFamilyNames;
+                THROW_IF_FAILED(fontFamily->GetFamilyNames(localizedFamilyNames.put()));
+
+                // use our current locale to find the correct name
+                BOOL exists{ FALSE };
+                UINT32 index;
+                HRESULT hr;
+                wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+                if (GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH))
+                {
+                    hr = localizedFamilyNames->FindLocaleName(localeName, &index, &exists);
+                }
+                if (SUCCEEDED(hr) && !exists)
+                {
+                    hr = localizedFamilyNames->FindLocaleName(L"en-us", &index, &exists);
+                }
+                if (!exists)
+                {
+                    // failed to find the correct locale, using the first one
+                    index = 0;
+                }
+
+                // get the localized name
+                UINT32 nameLength;
+                THROW_IF_FAILED(localizedFamilyNames->GetStringLength(index, &nameLength));
+                wchar_t* localizedName = new (std::nothrow) wchar_t[nameLength + 1];
```

Comment:
are we leaking this string? When we append it to the _FontList, won't we be making a new `hstring` to store the text, resulting in `localizedName` just leaking?

Reply:
The upcoming commit doesn't directly modify the font list. Instead, we're modifying a `std::vector<hstring>` then sorting/converting it over to the winrt-type when we're done. Does that fix this?

Reply:
No, that sounds completely unrelated ...

Reply:
You're allocating memory using new (instead of some WIL string wrapper) and never deleting it. First, **never allocate using `new/new[]` in this codebase**; second: if you _do_, you _must_ `delete/delete[]` it